### PR TITLE
adds expiration message for fcm provider

### PR DESCRIPTION
**Pull request by fanatic75**
## What does this PR do?

Adds expiration message for fcm provider

## Test Plan

(Write your test plan here. If you changed any code, please provide us with clear instructions on how you verified your changes work.)

## Related PRs and Issues

(If this PR is related to any other PR or resolves any issue or related to any issue link all related PR and issues here.)

### Have you read the [Contributing Guidelines on issues](https://github.com/utopia-php/messaging/blob/master/CONTRIBUTING.md)?

(Write your answer here.)

### CODE REVIEW

File: src/Utopia/Messaging/Adapter/Push/FCM.php [hunk 1]
```diff
@@ -144,10 +144,20 @@ protected function process(PushMessage $message): array
...
-                $result['response']['error']['message'] ?? ''
-            );
+
+            if (isset($result['response']['error'])) {
+                $response->addResultForRecipient(
+                    $message->getTo()[$index],
+                    $result['response']['error']['status'] === 'UNREGISTERED' ||
+                    $result['response']['error']['status'] === 'INVALID_ARGUMENT'
+                    ? 'Expired device token.'
+                    : $result['response']['error']['message'] ?? ''
```
Comment (stnguyen90) [inline src/Utopia/Messaging/Adapter/Push/FCM.php:154]: Why are we doing this and dropping the error message from the provider?

Comment (fanatic75) [reply src/Utopia/Messaging/Adapter/Push/FCM.php]: Each push type provider has a different exception message for when tokens expire. For push tokens, we are specifying an error message for expired device token. We decided to delete push targets which receive invalid token or expired token error. This error message has actually moved to Push Adapter base class. 

cc: @abnegate

File: src/Utopia/Messaging/Adapter/Push.php [hunk 2]
```diff
@@ -17,6 +17,11 @@ public function getMessageType(): string
         return PushMessage::class;
     }
 
+    protected static function getExpiredErrorMessage(): string
+    {
+        return 'Expired device token.';
+    }
```
Comment (stnguyen90) [inline src/Utopia/Messaging/Adapter/Push.php:23]: Why use a static function instead of a constant?

Comment (fanatic75) [reply src/Utopia/Messaging/Adapter/Push.php:23]: Following the same convention. 
It's same as getType method which returns a const string.

**Review by stnguyen90 — APPROVED** [review]

File: src/Utopia/Messaging/Adapter/Push.php (hunk 2 above)
Comment (eldadfux) [reply src/Utopia/Messaging/Adapter/Push.php:23]: @stnguyen90 message type and this can both return constants.

**Review by abnegate — APPROVED** [review]

**Pull request merged**